azure/datalake/store/transfer.py

```diff
@@ -240,9 +245,14 @@ def __init__(self, adlfs, transfer, merge=None, nthreads=None,
         self._chunked = chunked
         self._unique_temporary = unique_temporary
         self._unique_str = uuid.uuid4().hex
+        self._progress_callback=progress_callback
+        self._progress_lock = threading.Lock()
```

Comment:
@clehene this is wonderful and thank you so much for being so diligent in contributing to the repository, it is very appreciated!